web/portal/user/src/entities/Voicemail/Voicemail.tsx

```diff
@@ -1,25 +1,104 @@
-import { EntityValues } from '@irontec/ivoz-ui';
-import defaultEntityBehavior from '@irontec/ivoz-ui/entities/DefaultEntityBehavior';
-import EntityInterface from '@irontec/ivoz-ui/entities/EntityInterface';
+import { isEntityItem } from '@irontec/ivoz-ui';
+import defaultEntityBehavior, {
+  ChildDecorator as DefaultChildDecorator,
+} from '@irontec/ivoz-ui/entities/DefaultEntityBehavior';
+import EntityInterface, {
+  ChildDecoratorType,
+} from '@irontec/ivoz-ui/entities/EntityInterface';
 import _ from '@irontec/ivoz-ui/services/translations/translate';
-import AccountTreeIcon from '@mui/icons-material/AccountTree';
+import MailIcon from '@mui/icons-material/Mail';
 
-const VoiceMail: EntityInterface = {
+import {
+  VoicemailProperties,
+  VoicemailPropertyList,
+} from './VoicemailProperties';
+
+const properties: VoicemailProperties = {
+  enabled: {
+    label: _('Enabled'),
+    enum: {
+      '0': _('No'),
+      '1': _('Yes'),
+    },
+    default: '1',
+  },
+  name: {
+    label: _('Name'),
+    required: true,
+  },
+  sendMail: {
+    label: _('Voicemail send mail'),
+    enum: {
+      '0': _('No'),
+      '1': _('Yes'),
+    },
+    default: '1',
+    visualToggle: {
+      '0': {
+        show: [],
+        hide: ['attachSound', 'email'],
+      },
+      '1': {
+        show: ['attachSound', 'email'],
+        hide: [],
+      },
+    },
+  },
+  email: {
+    label: _('Email'),
+    required: true,
+  },
+  attachSound: {
+    label: _('Voicemail attach sound'),
+    enum: {
+      '0': _('No'),
+      '1': _('Yes'),
+    },
+    default: '1',
+  },
+};
+
+const columns = ['enabled', 'name', 'email'];
+
+export const ChildDecorator: ChildDecoratorType = (props) => {
+  const { routeMapItem, row } = props;
+
+  if (
+    isEntityItem(routeMapItem) &&
+    routeMapItem.entity.iden === voicemail.iden
+  ) {
+    const isDeletePath = routeMapItem.route === `${voicemail.path}/:id`;
+    const allowDelete = row.user === null && row.residentialDevice === null;
+
+    if (isDeletePath && !allowDelete) {
+      return null;
```

Comment:
```suggestion
      if (variant === 'text') {
        return <a className='disabled'>{voicemail.title}</a>;
      }

      return (
        <ChildEntityLink
          routeMapItem={routeMapItem}
          row={row}
          disabled={true}
        />
      );
```
Disable the button instead of hiding it